std/experimental/color/conv.d

```diff
@@ -0,0 +1,523 @@
+// Written in the D programming language.
+
+/**
+    This module implements various _color type conversions.
+
+    Authors:    Manu Evans
+    Copyright:  Copyright (c) 2015, Manu Evans.
+    License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+    Source:     $(PHOBOSSRC std/experimental/color/conv.d)
+*/
+module std.experimental.color.conv;
+
+import std.experimental.color;
+import std.experimental.color.rgb;
+import std.experimental.color.xyz;
+
+import std.traits: isNumeric, isIntegral, isFloatingPoint, isSigned, TemplateOf;
+import std.typetuple: TypeTuple;
+
+
+/**
+Convert between color types.
+*/
+To convertColor(To, From)(From color) if(isColor!To && isColor!From)
+{
+    // no conversion is necessary
+    static if(is(To == From))
+        return color;
+
+    // *** XYZ is the root type ***
+    else static if(isXYZ!From && isXYZ!To)
```

Comment:
> It's still an abomination.

no it is common practice with template constraints

> I don't want to do that to the symbol table.

that is no argument. are you written a linker or a maintainable, extensible library.

>  It wasn't an accident, it was entirely deliberate.
> I couldn't work out any way to make a fall-back constraint. 

which one is it. 

also when other color types get added. this function will become even bigger. That will not improve readability. 

but reviewing is not a zero sum game, so lets find a better solution.
I think we need multiple functions with small, readable template constraints.

you could actual partially specialize the templates
```d
T to(T)(RGB rbg) if(isXYZ!T) {}
```
that would make the template constraints really easy to read and you would have nice contained functions